src/Paystack/Gateway/PaystackGateway.php

```
@@ -0,0 +1,482 @@
+                    $donation->gatewayTransactionId
+                ),
+            ]);
+
+            return new PaymentRefunded();
+        } catch (PaymentGatewayException $exception) {
+            DonationNote::create([
+                'donationId' => $donation->id,
+                'content' => sprintf(
+                    __('Error! Donation %s was NOT refunded. Find more details on the error in the logs at Donations > Tools > Logs. To refund the donation, use the Paystack dashboard tools.', 'give-paystack'),
+                    $donation->id
+                ),
+            ]);
+
+            throw $exception;
+        }
+    }
+
+
+    /**
+     * @inheritDoc
+     */
+    public function enqueueScript(int $formId)
+    {
+        $scriptAsset = ScriptAsset::get(GIVE_PAYSTACK_URL . 'build/paystackGateway.asset.php');
+
+        // Enqueue our gateway script
+        wp_enqueue_script(
+            'give-paystack-gateway',
+            GIVE_PAYSTACK_URL . 'build/paystackGateway.js',
+            $scriptAsset['dependencies'],
+            $scriptAsset['version'],
+            true
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function formSettings(int $formId): array
+    {
+        return [
+            'publicKey' => $this->getPublicKey(),
+            'testMode' => give_is_test_mode(),
+            'formId' => $formId,
+        ];
+    }
+
+    /**
+     * Get the public key
+     *
+     * @since 3.0.0
+     */
+    public function getPublicKey(): string
+    {
+        return give_is_test_mode() ? give_get_option('paystack_test_public_key') : give_get_option('paystack_live_public_key');
+    }
+
+    /**
+     * Get the secret key
+     *
+     * @since 3.0.0
+     */
+    public function getSecretKey(): string
+    {
+        return give_is_test_mode() ? give_get_option('paystack_test_secret_key') : give_get_option('paystack_live_secret_key');
```

Should we throw an exception here in case nothing is found in the DB?

---

For this, instead of throwing an exception on the method itself I prefer to do that within context of usage.  So, I updated the implementation during createPayment to check if empty a597f0ed